Userland/Libraries/LibVideo/Containers/Matroska/MatroskaDemuxer.cpp

```diff
@@ -43,7 +43,7 @@ DecoderErrorOr<Vector<Track>> MatroskaDemuxer::get_tracks_for_type(TrackType typ
     Vector<Track> tracks;
     TRY(m_reader.for_each_track_of_type(matroska_track_type, [&](TrackEntry const& track_entry) -> DecoderErrorOr<IterationDecision> {
         VERIFY(track_entry.track_type() == matroska_track_type);
-        Track track(type, track_entry.track_number());
+        Track track(type, track_entry.track_number(), get_codec_id_for_string(track_entry.codec_id()));
```

Comment:
Let's make a `Demuxer::get_codec_for_track(Track track)` function so this can be a lazy getter instead of having to set the codec ID for tracks we may not use. Here, it's convenient since we already read out the string, but other containers may not want to do that ahead of time.

Later on, `get_codec_for_track()` may be used to request more track metadata if there is anything that we usually need around the same time that we need to get the track's codec (perhaps `TrackType`-specific information like nominal resolution).